src/client.rs

```diff
@@ -101,19 +95,19 @@ where
             let err = io::Error::new(io::ErrorKind::Other, "HTTPS scheme forced but can't be used");
             return HttpsConnecting(Box::new(future::err(err)));
         }
-        
         let host = dst.host().to_owned();
         let connecting = self.http.connect(dst);
-        let tls = self.tls.clone();
-        let verification = self.hostname_verification;
         let fut: BoxedFut<T::Transport> = if is_https {
+
+            let tls = match TlsConnector::builder()
```

Comment:
I haven't been following along, is the new way in native-tls 0.2 that you should build up a `TlsConnector` for every connect attempt?

Reply:
i did this because the only way to set danger_accept_invalid_hostnames is in the builder. A TlsConnector cannot be changed later, so it cant be shared.

I'm also not sure if the previous code was actually safe, since sharing ssl context with unrelated connections has side effects.

Reply:
I'd be fine with changing the mutator on `HttpsConnector` to be a constructor instead. But, I believe we still need to allow users to provide a `TlsConnector`, since that's the only way to allow them setup certificate chains or identities.

As for if it was unsafe: if it *was* unsafe, then definitely `TlsConnector::connect{_async}` should consume it, and it shouldn't implement `Clone`. I suspect that unsafety is actually abstracted away.

Reply:
Yeah, I'd have the user configure a TlsConnector and then pass that to the HttpsConnector.

It's safe to share a context between connections - that's the point of a context!